bat-utils/lib/runtime-queue.js

```diff
@@ -1,181 +1,80 @@
-const bluebird = require('bluebird')
+const BeeQueue = require('bee-queue')
 const redis = require('redis')
-const Rsmq = require('rsmq')
-const RsmqWorker = require('rsmq-worker')
 const SDebug = require('sdebug')
 const debug = new SDebug('queue')
-const underscore = require('underscore')
 
-bluebird.promisifyAll(redis.RedisClient.prototype)
-bluebird.promisifyAll(redis.Multi.prototype)
-
-const Queue = function (config, runtime) {
-  if (!(this instanceof Queue)) return new Queue(config, runtime)
-
-  if (!config.queue) return
-
-  this.config = config.queue.rsmq || config.queue
-  if (typeof this.config === 'string') {
-    if (this.config.indexOf('redis://') === -1) this.config = 'redis://' + this.config
-    this.config = { client: redis.createClient(this.config) }
-  }
-  this.rsmq = new Rsmq(this.config)
-  this.runtime = runtime
-
-  this.rsmq.on('connect', () => { debug('redis connect') })
-    .on('disconnect', () => { debug('redis disconnect') })
-    .on('error', (err) => {
-      debug('redis error', err)
-      this.runtime.captureException(err)
-    })
+Queue.prototype = {
+  send,
+  create,
+  register
 }
 
-Queue.prototype.create = async function (name) {
-  const self = this
-
-  return new Promise((resolve, reject) => {
-    self.rsmq.listQueues((err, rsp) => {
-      if (err) {
-        debug('listQueues failed')
-        return reject(err)
-      }
-      if (rsp.indexOf(name) !== -1) return resolve(false)
-
-      self.rsmq.createQueue({ qname: name }, (err, rsp) => {
-        if (err) {
-          debug('createQueue ' + name + ' failed')
-          return reject(err)
-        }
+module.exports = Queue
```

Comment:
I'm noticing that module.exports is defined in different places in different files throughout the codebase.  Ideally we can keep this consistent (e.g. always at the bottom of the file).  Where is the best place to have this line?

Reply:
So, i like to let all function declarations be hoisted up by js so that you can see a summary of the file's io quickly without having to look through the minutia of what the fns are actually doing. I believe marshal had previously set up the files to be definitions where no variable was available until it was declared as such. We can talk about it which to use and when in group dms

Reply:
I'm cool whatever you think is best, just hoping to be consistent